File: web-social-fe/lib/like-service.ts
import { api } from './api-service'

export interface LikeResponse {
  liked: boolean
  likesCount: number
}

export const LikeService = {
   toggleLikePost: async (postId: number): Promise<LikeResponse> => {
    const token = localStorage.getItem("token");
    const response = await api.post<LikeResponse>(
      `/likes/post/${postId}`, 
      {}, 
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      }
    )
    return response.data
  },

  toggleLikeComment: async (commentId: number): Promise<LikeResponse> => {
    const token = localStorage.getItem("token");
    const response = await api.post<LikeResponse>(
      `/likes/comment/${commentId}`, 
      {}, 
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      }
    )
    return response.data
  },

  checkPostLiked: async (postId: number): Promise<boolean> => {
    const response = await api.get<boolean>(`/likes/post/${postId}/status`)
    return response.data
  },
}

File: web-social/src/main/java/com/dran/web_social/services/LikeService.java
package com.dran.web_social.services;

import com.dran.web_social.dto.response.LikeResponse;

public interface LikeService {
    
    LikeResponse toggleLikePost(String username, Long postId);
    
    LikeResponse toggleLikeComment(String username, Long commentId);
    
    boolean isPostLikedByUser(Long postId, Long userId);
    
    boolean isCommentLikedByUser(Long commentId, Long userId);
    
    int getPostLikesCount(Long postId);
    
    int getCommentLikesCount(Long commentId);
}